### ISSUE TITLE
boolean options are not working

### ISSUE THREAD

**Issue by houyunf** [lexicalscope/jewelcli#3]
Any boolean (or Boolean) options are not working, it will always return true. The following code will not work.

@Option 
boolean getOption1()

@Option(defaultValue="false")
boolean getOption2()

If I did not pass any argument, both methods will return true. Your test case(http://jewelcli.sourceforge.net/xref-test/uk/co/flamingpenguin/jewel/cli/examples/TestPrimitiveExample.html) only tests when the boolean option arguments are present, however, there should also be test cases testing if no arguments are passed.

**Comment by houyunf** [comment]
forgot to mention that I am testing against 0.7.6.

**Comment by lexicalscope** [comment]
I am having trouble reproducing this behavior. Could you possibly post some code which illustrates how to recreate the problem? For example, what values are you passing to be parsed? 

Specifying defaultValue=false for a boolean option should not necessary, as missing boolean options should return false without any default value specified.

**Comment by houyunf** [comment]
Here's the test case. It will fail at last assert. I think the current implementation has problem,
1. enabled=true or enabled=false maybe not necessary, but should work anyway
2. if I define default value as true, then there is no way to set it to false.

# 

import junit.framework.TestCase;
import uk.co.flamingpenguin.jewel.cli.ArgumentValidationException;
import uk.co.flamingpenguin.jewel.cli.CliFactory;
import uk.co.flamingpenguin.jewel.cli.Option;

public class BooleanFlagTest extends TestCase {

  public interface Flags {

```
@Option
boolean getEnabled();
```

  }

  public void testFlags() throws ArgumentValidationException {
    // default
    Flags flags = CliFactory.parseArguments(Flags.class);
    assertFalse(flags.getEnabled());

```
// add boolean flag
flags = CliFactory.parseArguments(Flags.class, "--enabled");
assertTrue(flags.getEnabled());

// specify boolean value to true
flags = CliFactory.parseArguments(Flags.class, "--enabled", "true");
assertTrue(flags.getEnabled());

flags = CliFactory.parseArguments(Flags.class, "--enabled", "false");
// *** the test case failed here.
assertFalse(flags.getEnabled());
```

  }
}

**Comment by lexicalscope** [comment]
Ok, I understand.

The way the boolean options work is like this:

```
flags = CliFactory.parseArguments(Flags.class, "--enabled");
assertTrue(flags.getEnabled());
```

And:

```
flags = CliFactory.parseArguments(Flags.class);
assertFalse(flags.getEnabled());
```

The "true" and "false" are ignored. Boolean options don't take a value at the moment. If you want to do something similar to what you are describing you will have to define your own type:

```
public class Enable {
    public final boolean enable;
    public Enable(String string) {
        enable = Boolean.value(string);
    }
}
```

And define your class as: 

```
public interface Flags {
    @Option Enable getEnabled();
}
```

That would make it work like this

```
flags = CliFactory.parseArguments(Flags.class, "--enabled", "true");
assertTrue(flags.getEnabled().enabled);

flags = CliFactory.parseArguments(Flags.class, "--enabled", "false");
assertFalse(flags.getEnabled().enabled);
```

**Comment by houyunf** [comment]
Yes, I am currently coding like what you suggested, however, I think if will be supported natively, it will be simpler, and probably more consistent.

**Comment by lexicalscope** [comment]
Could you suggest how you would like it to work? I'm not sure exactly how to go about having a feature like that. Do you mean allowing an optional "true" and "false" for boolean options? Or something else?

**Comment by houyunf** [comment]
Yes, to make true and false work.

For example, for a flag "enabled",
1. no such flag: enabled = false, or whatever defined as default value
2. --enabled: enabled = true
3. --enabled false: enabled = false
4. --enabled true: enabled = true

Thanks.
